solutions/ex3/Main.hs

```diff
@@ -202,6 +216,22 @@ createSortInclusionInfer ''VarL    ''LhsL
 createSortInclusionInfer ''ExpL    ''RhsL
 createSortInclusionInfer ''AssignL ''StatementL
 
+--- Question: why is it enough to just convert AssignL to StatementL as sorts?
+-- An `Assign` constructor has more data than a `Statement` constructor.
+-- A `Block` constructor has more data than a `BlockStmt` constructor.
+-- I am assuming that the Template-generated code will automatically omit unnecessary data when converting an `Assign` to a `Statement`?
```

Comment:
This sounds like magical thinking, and there's a lot to unpack here.

You have the tools to investigate this kind of question yourself BTW.  Try running -ddump-splices.